packages/manager/modules/log-to-customer/src/list/controller.js

```diff
@@ -0,0 +1,164 @@
+import moment from 'moment';
+import { RETENTION } from '../constants';
+
+export default class LogToCustomerListCtrl {
+  /* @ngInject */
+  constructor($state, coreURLBuilder, LogToCustomerService, $translate, $q) {
+    this.$state = $state;
+    this.coreURLBuilder = coreURLBuilder;
+    this.LogToCustomerService = LogToCustomerService;
+    this.$translate = $translate;
+    this.$q = $q;
+  }
+
+  $onInit() {
+    this.loading = false;
+    this.accountList = [];
+    this.streams = [];
+    this.streamSubscriptions = {};
+    this.LDPHomePageLink = null;
+    this.selectedAccount = null;
+
+    this.LogToCustomerService.getIcebergLogAccounts().then((data) => {
+      this.accountList = data.map((item) => {
+        return {
+          ...item,
+          formattedName: item.displayName || item.serviceName,
+        };
+      });
+
+      this.selectedAccount =
+        this.accountList.length > 0 ? this.accountList[0] : null;
+      this.getStreamData();
+    });
+  }
+
+  getStreamData() {
+    this.$q
+      .all([this.getSubscribedStreams(), this.getLogAccountStreams()])
+      .then(() => {
+        this.loading = false;
+      });
+  }
+
+  getSubscribedStreams() {
+    this.streamSubscriptions = {};
+    return this.LogToCustomerService.icebergQuery(this.logSubscriptionUrl).then(
+      (data) => {
+        data.forEach((item) => {
+          this.streamSubscriptions[item.streamId] = { ...item };
+        });
+      },
+    );
+  }
+
+  createLogSubscription(streamId) {
+    this.loading = streamId;
```

Comment:
Why `streamId` instead of `true` ?

Reply:
Using a streamId, I can target in the datagrid which line is being updated and add a spinner on the corresponding button. Subscription's operations can take a few seconds, that way the user knows the process is running.